qcodes/dataset/plotting.py

```diff
@@ -252,17 +256,17 @@ def plot_dataset(dataset: DataSet,
         elif len(data) == 3:  # 2D PLOTTING
             log.debug(f'Doing a 2D plot with kwargs: {kwargs}')
 
-            # From the setpoints, figure out which 2D plotter to use
-            # TODO: The "decision tree" for what gets plotted how and how
-            # we check for that is still unfinished/not optimised
-
-            xpoints = flatten_1D_data_for_plot(data[0]['data'])
-            ypoints = flatten_1D_data_for_plot(data[1]['data'])
-            zpoints = flatten_1D_data_for_plot(data[2]['data'])
-
-            plottype = get_2D_plottype(xpoints, ypoints, zpoints)
-
-            log.debug(f'Determined plottype: {plottype}')
+            if data[2]["shape"] is None:
```

Comment:
I get a keyerror when trying out on a non-completed dataset. Changing the line to `if "shape" not in data[2].keys():` makes the PR work as intended. Note that this dataset was created by a manual measurement loop created locally. Data is fully grided so everything works but could be some way we set up the measurement context/loop that means shape was not created.

Reply:
Actually I think this is a combination of different bugs. Will have to do some more digging probably tomorrow to fully understand what is happening in the different cases.

It seems there are some issues around complex datatype. Maybe the shape information is not passed through correctly when getting real/imag parts for plotting. See for example `aaaaaaaa-0a00-0001-0000-01795d9a6abb` in DWH - this is a standard do2d-created dataset that was halted midway through. Dataset has two measured parameters - one of them being real and one of them complex. Real parameter plots fine but complex doesnt. But might actually be another bug in how the complex data is being exported.

My suggestion above definitely doesnt work in all cases.

Can probably dig some more tomorrow in this / do more testing if any updates are added in between.

Reply:
@ThorvaldLarsen This definitely was not handling complex plots correctly. I have fixed that now